packages/widgets/src/Pagination/Pagination.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react'
+import cx from 'classnames'
+import propTypes from '@dhis2/prop-types'
+import { colors, spacers } from '@dhis2/ui-constants'
+
+import { PageSizeSelect } from './PageSizeSelect.js'
+import { PageSummary } from './PageSummary.js'
+import { PageSelect } from './PageSelect'
+import { PageControls } from './PageControls'
+
+import { sectionStyles } from './Pagination.styles.js'
+
+// TODO: discuss padding with Joe and Ismay
+
+/**
+ * @module
+ * @param {Pagination.PropTypes} props
+ *
+ * @returns {React.Component}
+ *
+ * @example import { Pagination } from @dhis2/ui
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/molecules/pagination.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/pagination--default|Storybook}
+ */
+const Pagination = ({
+    className,
+    dataTest,
+    hidePageSizeSelect,
+    hidePageSelect,
+    page,
+    pageCount,
+    pageSize,
+    total,
+    pageSizes,
+    onPageChange,
+    onPageSizeChange,
+}) => {
+    return (
+        <div className={cx('container', className)} data-test={dataTest}>
+            {hidePageSizeSelect ? (
+                <div className="spacer" />
+            ) : (
+                <PageSizeSelect
+                    dataTest={dataTest}
+                    pageSize={pageSize}
+                    pageSizes={pageSizes}
+                    onChange={onPageSizeChange}
+                />
+            )}
+            <PageSummary
+                dataTest={dataTest}
+                page={page}
+                pageCount={pageCount}
+                pageSize={pageSize}
+                total={total}
+            />
+            <div className="section">
```

Comment:
Could/should be its own component (e. g. `PaginationSection`) which would reduce the complexity of importing the section styles in multiple places which causes multiple style tags and adheres to the styled-jsx philosophy

Reply:
I liked this suggestion, but when I tried implementing it I ran into some issues with overriding the styles for the `SingleSelect`s using `styled-jsx`. Also, there were only three sections and each one is slightly different. Long story short: in the end I duplicated a few lines of CSS in 3 components and got rid of the shared styles instead, see e03f38b.

I'm resolving this comment, because I already discussed this briefly on Slack

Reply:
> each one is slightly different.

That also makes me prefer the direction Hendrik took here 👍